### DISCUSSION TITLE
Creating an environment from a PandaPower format schema

### DISCUSSION THREAD

**Discussion by epetrishchev** [rte-france/Grid2Op#288] [Q&A]
Hello. I have such a question. Can I upload my PandaPower format schema as an environment? If so, how can I do it. Thank you very much.

**Comment by BDonnot** [comment]
Hello,

I'm not sure i master pandapower that well, so i'm not sure I can answer the question correctly. Might first guess would be:
"yes you probably can" (I assume that you have generation and loads for all generators and all consumptions and all timesteps)

Can you show me what a `PandaPower format schema` looks like ? I might be able to help with an example :-)

Also, you might want to look here https://grid2op.readthedocs.io/en/latest/available_envs.html#content-of-an-environment and there https://grid2op.readthedocs.io/en/latest/available_envs.html#possible-workflow-to-create-an-environment-from-existing-chronics if you want some generic information about a grid2op environment (which might be handy :-) )

**Reply by epetrishchev** [reply]
Thanks for the quick response. Yes, I have read the documentation.  Here is my power grid in xlsx format (unfortunately, the transfer of a JSON file is not supported by GitHub). It is calculated correctly in PandaPower. The question is most likely in creating data for the "chronics" folder. After all, my power system is data for the current mode and I don't quite understand how I can generate data for different time intervals. Of course, I can just change them manually, well, for example, slightly change the power consumption of any load, but then how to save this state of the power system as a state for the next time interval?
[grid.xlsx](https://github.com/rte-france/Grid2Op/files/8115493/grid.xlsx)

**Reply by BDonnot** [reply]
Oh ok I see.

So you have a powergrid, and then you would like to come up with some kind of loads and productions to apply to your grid ?

If that's so, the process  is not automatic at all, and there is no unique answer to that.

My first advice would be for you to check if there exists already some available temporal data for your grid. If so, you can use them. If not, you will have to invent them somehow.

One way is to start from existing data (the "reference data"). Then you can scale up the loads and apply a repartition keys to modify the loads in your grid accordingly. Basically, each loads in your grid will be proportional to the total demand of reference data. I think it's best to add some noise to avoid correlation of 1.0 between your loads.

Then you need to come up with a solution for renewable energy sources. You can here use the same technique as above.

And then, this is the most difficult part, you need to come up with a solution to generate the controlable generators (hydro, nuclear, coal, gaz etc.) This can be done with a kind of market design / economic dispatch / opf kind of model.
